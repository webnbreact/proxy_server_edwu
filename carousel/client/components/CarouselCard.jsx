import React from 'react';

import { Image, OverFlowWrap } from '../assets/StyledAssets.jsx'
// import styled, { css } from 'styled-components';

// const Image = styled.div`

//   // border: 3px solid green;
//   background: no-repeat url(${({ imageUrl }) => imageUrl});
//   background-size: cover;
//   background-position: center;
//   min-width: 100%;
//   min-height: 100%;
//   width: 500px;
//   height: 300px;
//   ${({ highlightedPicIdx, currPicIdx }) => (highlightedPicIdx !== currPicIdx) && css`
//     -webkit-filter: brightness(50%);
//     // border: 5px solid red;
//     transition: all .8s ease-out;
//   `}


//   ${({ inModal }) => inModal && css`
//   : hover {
//   color: #424242;
//   background-size: cover;
//   -webkit-transition: all .3s ease-out;
//   -moz-transition: all .3s ease-out;
//   -ms-transition: all .3s ease-out;
//   -o-transition: all .3s ease-out;
//   transition: all .3s ease-out;
//   opacity: 1;
//   transform: scale(1.3);
//   -ms-transform: scale(1.3); /* IE 9 */
//   -webkit-transform: scale(1.3); /* Safari and Chrome */
//   // border: 3px solid blue;

//   `}
// `;

// const OverFlowWrap = styled.div`
//   position: relative;
//   overflow: hidden;
//   // border: 1px solid transparent;
//   width: 100 %;
//   height: 100 %;

//   ${({ inModal }) => !inModal && css`
//     // border: 3px solid red;
//     width:100%;
//     height:100%;
//   `}
// `;

const CarouselCard = (props) => {
  const { highlightedPicIdx, handleHover, inModal, imageUrl, currPicIdx, handleModalClick } = props;
  return (
    <OverFlowWrap inModal={inModal}>
      <Image
        highlightedPicIdx={highlightedPicIdx}
        onFocus={(e) => { handleHover(e, currPicIdx); }}
        onMouseOver={(e) => { handleHover(e, currPicIdx); }}
        inModal={inModal}
        onClick={(e) => { handleModalClick(e, currPicIdx); }}
        currPicIdx={currPicIdx}
        imageUrl={imageUrl}
        key={currPicIdx}
        alt="pictures"
      />
    </OverFlowWrap>
  );
};

export default CarouselCard;
