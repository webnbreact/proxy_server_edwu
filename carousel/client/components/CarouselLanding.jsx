/* eslint-disable no-useless-constructor */
/* eslint-disable class-methods-use-this */
import React from 'react';
import axios from 'axios';
import url from 'url';
import styled from 'styled-components';
// import { BrowserRouter as Brouter, Switch, Route, Link } from 'react-router-dom';
import { withRouter } from 'react-router';
import CarouselCard from './CarouselCard';

const LandingPhotoGrid = styled.div`
  display: grid;
  grid-template-columns: auto 40vw 40vw;
  grid-auto-rows: 50vh;
  border: 3px solid purple;
  width:98vw;
  margin-right: 1vw;
`;

const MainLandingPhotoStyled = styled.div`
  display: grid;
  grid-column: 1/3;
  gap: 0px;
`;

const SideLandingPhotoGrid = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  border:1px solid transparent;
  overflow:hidden;
  border: 5px solid blue;
`;

class CarouselLanding extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      imageUrls: [],
      highlightedPicIdx: 0,
    };

    this.handleHover = this.handleHover.bind(this);
  }

  componentDidMount() {
    let { location, match, history, handleGetPictures } = this.props;

    const dbLocation = 'http://localhost:4500';
    const postUrl = url.resolve(dbLocation, location.pathname);

    this.getDataFromServer(postUrl)
      .then((data) => {
        this.setState({ imageUrls: data });
        handleGetPictures(data);
      });
  }

  getDataFromServer(postUrl) {
    return axios.get(postUrl)
      .then(response => (response.data));
  }

  handleHover(e, highlightedPicIdx) {
    this.setState({ highlightedPicIdx: highlightedPicIdx });
  }

  render() {
    const { imageUrls } = this.state;
    const { handleModalClick } = this.props;
    const { highlightedPicIdx } = this.state;
    return (
      <LandingPhotoGrid>
        <MainLandingPhotoStyled>
          <CarouselCard
            inModal
            handleModalClick={handleModalClick}
            handleHover={this.handleHover}
            highlightedPicIdx={highlightedPicIdx}
            currPicIdx={0}
            key={0}
            imageUrl={imageUrls[0]}
          />
        </MainLandingPhotoStyled>
        <SideLandingPhotoGrid>
          {
            imageUrls.map((imageUrl, index) => {
              return (
                <CarouselCard
                  highlightedPicIdx={highlightedPicIdx}
                  handleHover={this.handleHover}
                  inModal
                  currPicIdx={index}
                  handleModalClick={handleModalClick}
                  key={index}
                  imageUrl={imageUrl}
                />
              );
            }).slice(1, 5)
          }
        </SideLandingPhotoGrid>
      </LandingPhotoGrid>
    );
  }
}


export default withRouter(CarouselLanding);
