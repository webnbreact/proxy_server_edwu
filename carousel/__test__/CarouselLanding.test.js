import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import CarouselLanding from '../client/components/CarouselLanding';
import CarouselCard from '../client/components/CarouselCard';

Enzyme.configure({ adapter: new Adapter() });

describe('Carousel Landing', () => {
  it('should exist exist', () => {
    const tree = shallow(<CarouselLanding />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('should render four Carousel Card components', () => {
    const tree = shallow(<CarouselLanding />);
    tree.setState({
      imageUrls: [
        'http://google.com',
        'http://google.com',
        'http://google.com',
        'http://google.com',
      ],
    });
    expect(tree.find(CarouselCard).length).toEqual(4);
  });
});

// test('This tests how many nodes are on the screen', () => {
//   const tree = shallow(<CarouselLanding />);
// });
