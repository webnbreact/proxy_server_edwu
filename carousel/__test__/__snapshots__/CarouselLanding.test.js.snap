// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Landing should exist exist 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Carousel Landing to exist 1`] = `<div />`;
